modules/woocommerce-analytics/classes/wp-woocommerce-analytics-universal.php

```diff
@@ -0,0 +1,355 @@
+	 */
+	public function __construct() {
+		// loading _wca
+		add_action( 'wp_head', array( $this, 'wp_head_top' ), 1 );
+
+		// loading s.js
+		add_action( 'wp_head', array( $this, 'wp_head_bottom' ), 999999 );
+
+		// single product page view
+		add_action( 'woocommerce_after_single_product', array( $this, 'product_detail' ) );
+
+		// add to cart on single product page
+		add_action( 'woocommerce_after_add_to_cart_button', array( $this, 'add_to_cart' ) );
+
+		// add to carts from list views (search, store etc.)
+		add_action( 'wp_footer', array( $this, 'loop_add_to_cart' ) );
+
+
+		add_action( 'woocommerce_after_cart', array( $this, 'remove_from_cart' ) );
+		add_action( 'woocommerce_after_mini_cart', array( $this, 'remove_from_cart' ) );
+		add_action( 'wcct_before_cart_widget', array( $this, 'remove_from_cart' ) );
+		add_filter( 'woocommerce_cart_item_remove_link', array( $this, 'remove_from_cart_attributes' ), 10, 2 );
+
+		// cart checkout
+		add_action( 'woocommerce_after_checkout_form', array( $this, 'checkout_process' ) );
+
+		// order confirmed
+		add_action( 'woocommerce_thankyou', array( $this, 'order_process' ), 10, 1 );
+		add_action( 'woocommerce_after_cart', array( $this, 'remove_from_cart_via_quantity' ), 10, 1 );
+	}
+
+	/**
+	 * Make _wca available to queue events
+	 */
+	public function wp_head_top() {
+
+		if ( is_admin() ) {
+			return;
+		}
+
+		if ( ! class_exists( 'WooCommerce' ) ) {
+			return;
+		}
+
+		$wca_code = "<script>window._wca = window._wca || [];</script>";
+		echo "$wca_code\r\n";
+	}
+
+
+	/**
+	 * Place script to call s.js, Store Analytics
+	 */
+	public function wp_head_bottom() {
+
+		if ( is_admin() ) {
+			return;
+		}
+
+		if ( ! class_exists( 'WooCommerce' ) ) {
+			return;
+		}
+
+		$async_code = "<script async src='https://stats.wp.com/s.js'></script>";
+		echo "$async_code\r\n";
+	}
+
+	/**
+	 * On a product page, add a click event listener to "Add to Cart" button click
+	 */
+	public function add_to_cart() {
+
+		if ( ! is_single() ) {
+			return;
+		}
+
+		$blogid = Jetpack::get_option( 'id' );
+		global $product;
+
+		$selector = '.single_add_to_cart_button';
+
+		wc_enqueue_js(
+			"jQuery( '" . esc_js( $selector ) . "' ).click( function() {
+				_wca.push( {
+					'_en': 'woocommerceanalytics_add_to_cart',
+					'blog_id': " . esc_js( $blogid ) . ",
+					'pi': '" . esc_js( $product->get_id() ) . "',
+					'pn' : '" . esc_js( $product->get_title() ) . "',
+					'pq': jQuery( 'input.qty' ).val() ? jQuery( 'input.qty' ).val() : '1',
+					'ui': '" . esc_js( $this->get_user_id() ) . "',
+				} );
+			} );"
+		);
+	}
+
+	/**
+	 * On product lists or other non-product pages, add an event listener to "Add to Cart" button click
+	 */
+	public function loop_add_to_cart() {
+
+		if ( ! class_exists( 'WooCommerce' ) ) {
+			return;
+		}
+
+		$minimum_woocommerce_active = class_exists( 'WooCommerce' ) && version_compare( WC_VERSION, '3.0', '>=' );
```

Comment:
Is the version check here due to the selector not being present until that version? Also the check for `WooCommerce` already happened so not sure it needs repeating here.